### DISCUSSION TITLE
Can I use .env.development as the storage of environment variables when I build

### DISCUSSION THREAD

**Discussion by Lrunlin** [vercel/next.js#40009] [Help]
![image](https://user-images.githubusercontent.com/54397964/187031061-1bcc3ce4-e2f9-4f7f-ba1d-ebf1df6eb2c1.png)
I tried this method, but unfortunately, it didn't work
![image](https://user-images.githubusercontent.com/54397964/187031080-3ed807bb-675a-42de-a8d2-ef800a7a6e84.png)

**Comment by icyJoseph** [comment]
The https://nextjs.org/docs/basic-features/environment-variables#environment-variable-load-order page, shows that the `.env` files are read in an specific order, and depending on the `NODE_ENV` variable, some are picked up while others are ignored.

When you do `next build`, then `NODE_ENV` is set to `production`, so `.env.production` is read as part of the ENV vars loading process.

Meanwhile, `.env.development` is ignored because `NODE_ENV` is set to `production`. In `next dev`, `NODE_ENV` is set to `development`.

**Reply by Lrunlin** [reply]
Thank you for your reply. I have solved this problem. Use the dotenv plug-in to read the. Env.xxx file and set it in next.config JS env attribute. Cross env node during build_ Env = development set environment variable
![image](https://user-images.githubusercontent.com/54397964/187171322-596e854c-f3ef-42e7-a060-a6b752dc4236.png)
![image](https://user-images.githubusercontent.com/54397964/187171355-7562e789-9009-436b-8e23-1048a401243e.png)

**Reply by Lrunlin** [reply]
> 感谢你的回复。我已经解决了这个问题。使用 dotenv 插件读取。Env.xxx 文件并将其设置在 next.config JS env 属性中。build 期间跨 env 节点_Env = 开发集环境变量 ![图片](https://user-images.githubusercontent.com/54397964/187171322-596e854c-f3ef-42e7-a060-a6b752dc4236.png) ![图片](https://user-images.githubusercontent.com/54397964/187171355-7562e789-9009-436b-8e23-1048a401243e.png)

Chinese：
谢谢您的回复，我已经解决了这个问题。使用dotenv插件读取.env.XXX文件，后设置在next.config.js的env属性中。在build时cross-env NODE_ENV=development设置环境变量
